components/console/commands.c

```diff
@@ -257,7 +262,7 @@ esp_err_t esp_console_run(const char *cmdline, int *cmd_ret)
         *cmd_ret = (*cmd->func)(argc, argv);
     }
     if (cmd->func_w_context) {
-        *cmd_ret = (*cmd->func_w_context)(cmd->context, argc, argv);
+        *cmd_ret = (*cmd->func_w_context)(invocation_ctx, argc, argv);
```

Comment:
```suggestion
        *cmd_ret = (*cmd->func_w_context)(invocation_ctx ?: cmd->context, argc, argv);
```
I think this is what you meant, else the commands will never ne executed with their default context